src/Modules/UngradedModules/FreeModElem.jl

```diff
@@ -216,14 +216,26 @@ function (==)(a::AbstractFreeModElem, b::AbstractFreeModElem)
   return a.coords == b.coords
 end
 
-function hash(a::AbstractFreeModElem, h::UInt)
+function hash(a::AbstractFreeModElem{<:MPolyElem}, h::UInt)
   b = 0xaa2ba4a32dd0b431 % UInt
   h = hash(typeof(a), h)
   h = hash(parent(a), h)
   h = hash(coordinates(a), h)
   return xor(h, b)
 end
 
+function hash(a::AbstractFreeModElem{<:MPolyQuoRingElem}, h::UInt)
+  simplify!(a)
```

Comment:
We just discussed how this call resp. this `hash` method is somewhat dangerous:
- it assumes that `simplify!(a)` modifies `a` in-place and also "remembers" that it is simplified (so it is fast "amortized", i.e. you only compute a GB for it once)
- it must produce a unique normal form -- but we don't really have a specification for `simplify!`. We really need one otherwise we are building on quicksand
- perhaps it would be better to just do `error("not supported")` here and handle using these in dicts differently??
- however the old `hash(a::AbstractFreeModElem, h::UInt)` method just was wrong...